### PR TITLE
set minimal HIP version to 3.5

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- Update to the FindHIP implementation from rocm-3.5
- set HIP 3.5 as minimum requirement
- switch CI base container for HIP to rocm3.5 container
  - test HIP-clang instead of HIP-nvcc
  - remove testing for HIP-nvcc

### CODE REVIEW

**Review by BenjaminW3 — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -74,10 +74,10 @@ jobs:
         include:
         ## HIP
         - name: linux_hip_nvcc-9.2_gcc-5_debug_hip_only
-          os: ubuntu-latest
+          os: rocm/pytorch:rocm3.5_ubuntu16.04_py3.6_pytorch
```
Comment (psychocoderHPC) [inline .github/workflows/ci.yml]: @BenjaminW3 I expect this is the place to load a different base container or? I will switch to a container with clean installed hip-clang.

Comment (BenjaminW3) [reply .github/workflows/ci.yml]: I do not think that this works. The `os:` only allows one of the few virtual machine types which github allows.

Comment (BenjaminW3) [reply .github/workflows/ci.yml]: `ALPAKA_CI_DOCKER_BASE_IMAGE_NAME` is the correct one

Comment (psychocoderHPC) [reply .github/workflows/ci.yml]: thx

**Comment by psychocoderHPC** [comment]
@BenjaminW3 I switched now to a rocm container and we test HIP-clang. HIP-clang was always the target for HIP and since rocm3.5 is shipping HIP-clang we do not handle installation.

File: script/install.sh [hunk 2]
```diff
@@ -67,7 +66,7 @@ fi
 if [ "$ALPAKA_CI_OS_NAME" = "Linux" ]
 then
     # Minimize docker image size
-    sudo apt-get --quiet --purge autoremove
+    sudo apt-get --quiet --purge --yes autoremove
     sudo apt-get clean
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 fi
```
Comment (BenjaminW3) [inline script/install.sh]: You can delete this whole `if-fi` block because it is not useful as we do not cache the image anymore.

Comment (psychocoderHPC) [reply script/install.sh]: will do

**Comment by psychocoderHPC** [comment]
I removed the if block and CI passed, ready to merge

**Review by BenjaminW3 — APPROVED** [review]

**Pull request merged**